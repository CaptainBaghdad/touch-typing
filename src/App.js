//import React, { Component } from 'react';
import * as React from 'react';
import logo from './logo.svg';
import './App.css';
//
import TextField from 'material-ui/TextField';
import Keyboard from 'react-material-ui-keyboard';
import { alphaNumericKeyboard } from 'react-material-ui-keyboard/layouts';

class App extends React.Component {

  state = {
    open:false,
    value: ''

  }

  handleInput  = (input) => {
    this.setState({value: input})

  }


  render() {

    
    return (
      <div>
        <h1>Press the area below this is where the keyboard is</h1>
      <Keyboard
        textField={
          <TextField
            id="text"
            value={this.state.value}
          />
        }
        automatic
        onInput={this.handleInput}
        layouts={[alphaNumericKeyboard]}
      />

      
      </div>
        
    );
  }
}

export default App;
