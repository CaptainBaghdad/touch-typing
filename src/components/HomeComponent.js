//import React, { Component } from 'react';
import * as React from 'react';
import { Link } from 'react-router-dom';
import FooterComponent from  '../components/FooterComponent';

//


import '../css/home.css'
class HomeComponent extends React.Component {

  state = {
    open:true,
    value: '',
    green: false,
    red: false,
    correct: [],
    wrong: [],
    start: 0,
    name: '',
    email: '',
    password:''


  }

  
  

  handleChange = (e) => {
    //console.log(`this is from the change ${e.target.value}`);
   this.setState({value: e.target.value }) ;

  }


  
  


  render() {

    
    return (
      <div className="container">
        
        <div id="section-1-r">
      <Link to="/register" className="links">Login
      </Link>

      <Link to="/login" className="links" >Register</Link>
        </div>


        <div className="main-title">
        <h3>Touch Typing</h3>

        </div>
       


  <section id="first-home-section">
  <img src={process.env.PUBLIC_URL + '/images/typing.jpg'}  id="main-img" alt="No photo found" />
      <br/>
      <center>
      <h3>Learn touch typing</h3>
      <br/>
      <br/>

      <button className="btn btn-success">Try it</button>
      </center>
  </section>

    <section id="second-home-section">
    <center>choose one
    <ul>
      <li><Link to="/letteru"><img className="display-pic" src={process.env.PUBLIC_URL + '/images/letter-u.jpg'} alt="no pic" /></Link></li>
      <li><Link to="/letterr"><img className="display-pic" src={process.env.PUBLIC_URL + '/images/letter-r.png'} alt="mp pic" /></Link></li>
      <li><Link to="/letteri"><img className="display-pic" src={process.env.PUBLIC_URL + '/images/letter-i.jpg'} alt="no pic" /></Link></li>
      <li><Link to="/lettere"><img className="display-pic" src={process.env.PUBLIC_URL + '/images/letter-e.png'} alt="no pic" /></Link></li>
    </ul>


    </center>
    </section>


       <section id="footer-holder">
         <FooterComponent />
       </section>
          
      
    </div>
      
      
        
    );
  }
}

export default HomeComponent;
