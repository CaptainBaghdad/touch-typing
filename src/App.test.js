import React from 'react';
import ReactDOM from 'react-dom';
import { create } from "react-test-renderer";
import HomeComponent from "./components/Home.Component";

import App from './App';

describe("Home Component", () => {
  test('it matches the snapshot', () => {
    const component = create(<HomeComponent />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
