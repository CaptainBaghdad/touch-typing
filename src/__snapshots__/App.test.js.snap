// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component it matches the snapshot 1`] = `
<div>
  <p>
    This is the home component
  </p>
</div>
`;
